### ISSUE TITLE
Repeated Flag_... branches in some productions

### ISSUE THREAD

**Issue by arizzi** [cms-nanoAOD/cmssw#94]
original mail from @mbluj  

Hi Andrea,

On Fri, Jan 12, 2018 at 4:34 PM, Michal Bluj <michal.bluj@gmail.com> wrote:

On Fri, Jan 12, 2018 at 4:27 PM, Andrea Rizzi <andrea.rizzi@cern.ch> wrote:
Hi,
yes go first to cms-nanoAOD.

80X NANOAOD private-crab production is going already available (see
the WorkBookNanoAOD twiki) and first centrally produced samples are
also available here:
https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=dataset+dataset%3D%2F*%2F*05Jan*%2FNANOAOD*+status%3D*
OK, great. Is the campaign i.e. list of samples described somewhere?
I have checked one file from this production (*) and I see an issue with duplicated/hidden branches (**) related to filter flags (21 flags of type Boot_t), e.g. Flag_HBHENoiseFilter, etc. It is possible to draw values of the flags with interactive root, but it can be a sign of some issue, e.g. that some branches are created and written twice. 
I do not see the issue with samples I produced with genuine CMSSW_9_4_2 using 80X MC as input (with crab). 

Best,
Michał

(*) https://cmsweb.cern.ch/das/request?input=file%3D%2Fstore%2Fdata%2FRun2016D%2FSingleMuon%2FNANOAOD%2F05Jan2018-v1%2F10000%2F0C7F667D-33F6-E711-9F47-7845C4FC379D.root&instance=prod%2Fglobal
(**) An error message returned by MakeClass tool: "Error in <TTreePlayer::MakeClass>: The branch named Flag_HBHENoiseFilter (full path name: Flag_HBHENoiseFilter) is hidden by another branch of the same name and its data will not be loaded."

**Comment by mbluj** [comment]
Hello,

for completeness: here you can find example of files with my private production with master of PhysicsTools/NanoAOD on top on 9_4_2 (*) with 80X MC as an input. The files do not consist duplicated branches. The production was performed with crab using one thread mode.

(*) https://cmsweb.cern.ch/das/request?input=file%20dataset%3D/VBFHToTauTau_M125_13TeV_powheg_pythia8/bluj-RunIISummer16NanoAOD940-PUMoriond17_80X_mcRun2_asymptotic_2016_TrancheIV_v6-v4-d92fa2c3e8a61d39c849d345172d3d98/USER&instance=prod/phys03&idx=0&limit=10

**Comment by gpetruc** [comment]
I think the issue may be that we make flags from all TriggerResults objects, and in re-miniAOD campaigns we sometimes end up having both the old and the new TriggerResults in the event.
Can someone check if this is the source of the duplication?

**Comment by aminnj** [comment]
Hi,

I recently ran into this after trying to run some tools on a file from /DoubleMuon/Run2017E-31Mar2018-v1/NANOAOD [1]. It's not really an issue since the content is identical, but it might break some user tools since branch names are the same.

Has this been fixed since that dataset was produced?

Thanks!
Nick

[1]

```python
import ROOT as r
from collections import Counter

f1 = r.TFile.Open("root://cmsxrootd.fnal.gov//store/data/Run2017E/DoubleMuon/NANOAOD/31Mar2018-v1/100000/585C6E66-206C-E811-AE6C-90E2BAD1C9A8.root")
t = f1.Get("Events")

freq = Counter(b.GetName() for b in list(t.GetListOfBranches()))
print [name for name,count in freq.items() if count > 1]
``

**Comment by PerilousApricot** [comment]
Hello,

This issue breaks the `spark-root` adaptor in a subtle way, that will be a difficult fix. Is there a workaround to somehow strip the duplicate branches during production, or is a post-processing step the only solution currently?

**Comment by arizzi** [comment]
I will have a look later this week.

Il sab 25 ago 2018, 07:26 Andrew Melo <notifications@github.com> ha scritto:

**Comment by arizzi** [comment]
the Flag duplication seem to be to the fact that we run them both in RECO and PAT ... I'm not sure what is the right thing to do here

**Comment by PerilousApricot** [comment]
would it make sense to perhaps namespace them as `flag_reco_*` and `flag_pat_*`?

**Comment by peruzzim** [comment]
fixed in #478
